float64 min_x
float64 max_x
float64 min_y
float64 max_y
float64 x
float64 y
float64 r
---
int32 num_of_obstacle_point 
